include/materials/norsand.tcc

```diff
@@ -173,13 +147,13 @@ void mpm::NorSand<Tdim>::compute_state_variables(
     mpm::norsand::FailureState yield_type) {
 
   // Get invariants
-  auto invariants = this->compute_stress_invariants(stress, state_vars);
+  auto* invariants = this->compute_stress_invariants(stress, state_vars);
```

Comment:
This auto* means we have no idea how big an array we are returning. Please pass as in-out arguments